docs: Explain remembered space behavior (#6937) (#7418)
elastic/docs-content: deploy-manage/choose-and-switch-spaces.md

---
description: Select and switch Kibana spaces, and control whether Kibana returns to your last space when you log in.
applies_to:
  stack: ga
  serverless: ga
products:
  - id: kibana
  - id: cloud-serverless
---

# Select and switch Kibana spaces

The spaces you can access depend on your roles. Switching spaces does not change your permissions.

## Switch spaces

From the global header, select your current space, then select the space you want to open.

:::{image} /deploy-manage/images/kibana-change-space.png
:alt: Change current space menu
:screenshot:
:width: 50%
:::

## Understand what happens when you log in

> {applies_to}`ech: ga` {applies_to}`ece: ga` {applies_to}`eck: ga` {applies_to}`serverless: ga` If you can access more than one space, {{kib}} asks you to select a space when you log in.

Reviewer: this split (ech/ece/eck vs. self) makes no sense to me. this feels like more of a stack version thing - at minimum, cloud deployments vs. self-managed ones. ece/eck definitely don't have explicit control over kibana behavior

Author: Yes, this is all being checked because I talked to the product team a few days ago and it seems they intended this to work on all deployment types initially. I'll return this to draft

Author: ```suggestion
{applies_to}`ech: ga` {applies_to}`serverless: ga` If you can access more than one space, {{kib}} asks you to select a space when you log in.
```
Fix

> {applies_to}`self: ga` On self-managed deployments, what {{kib}} opens when you log in depends on your version and whether it has remembered a space:

Author: ```suggestion
{applies_to}`self: ga` {applies_to}`ece: ga` {applies_to}`eck: ga` On self-managed deployments, what {{kib}} opens when you log in depends on your version and whether it has remembered a space:
```

- {applies_to}`self: ga 9.5+` {{kib}} can remember the space you use and return you to it. If {{kib}} has no remembered space or you no longer have access to it, {{kib}} asks you to select a space. A direct link continues to open the linked {{kib}} app or object instead of redirecting you to the remembered space.

> - {applies_to}`self: ga 9.0-9.4` If you can access more than one space, {{kib}} asks you to select a space.

Author: ```suggestion
{applies_to}`self: ga` On self-managed deployments, what {{kib}} opens when you log in depends on your version and whether it has remembered a space:

   - {applies_to}`self: ga 9.5+` {{kib}} can remember the space you use and return you to it. If {{kib}} has no remembered space or you no longer have access to it, {{kib}} asks you to select a space. A direct link continues to open the linked {{kib}} app or object instead of redirecting you to the remembered space.
   - {applies_to}`self: ga 9.0-9.4` If you can access more than one space, {{kib}} asks you to select a space.
```

>    - {applies_to}`self: ga 9.0-9.4` If you can access more than one space, {{kib}} asks you to select a space.

Author: ```suggestion
   - {applies_to}`stack: ga 9.5+` {{kib}} can remember the space you use and return you to it. If {{kib}} has no remembered space or you no longer have access to it, {{kib}} asks you to select a space. A direct link continues to open the linked {{kib}} app or object instead of redirecting you to the remembered space.
   - {applies_to}`stack: ga 9.0-9.4` If you can access more than one space, {{kib}} asks you to select a space.
```

## Change whether Kibana remembers your space
```{applies_to}
deployment:

>   self: ga 9.5+

Author: ```suggestion
  self: ga
  ece: ga
  eck: ga
```

```

This personal preference controls whether {{kib}} returns you to your last space or shows the space selector when you log in and can access multiple spaces.

> 

Author: ```suggestion

**Requirements**:
- Your Kibana instance must be self-managed or deployed through {{ece}} or {{eck}}. This menu option is not available on {{ecloud}}.

```

## Switch spaces

From the global header, select your current space, then select the space you want to open.

:::{image} /deploy-manage/images/kibana-change-space.png
:alt: Change current space menu
:screenshot:
:width: 50%
:::

## Understand what happens when you log in

{applies_to}`ech: ga` {applies_to}`serverless: ga` If you can access more than one space, {{kib}} asks you to select a space when you log in.

{applies_to}`self: ga` {applies_to}`ece: ga` {applies_to}`eck: ga` On self-managed deployments, what {{kib}} opens when you log in depends on your version and whether it has remembered a space:

   - {applies_to}`stack: ga 9.5+` {{kib}} can remember the space you use and return you to it. If {{kib}} has no remembered space or you no longer have access to it, {{kib}} asks you to select a space. A direct link continues to open the linked {{kib}} app or object instead of redirecting you to the remembered space.
   - {applies_to}`stack: ga 9.0-9.4` If you can access more than one space, {{kib}} asks you to select a space.

## Change whether Kibana remembers your space
```{applies_to}
deployment:
  self: ga
  ece: ga

>   eck: ga

Author: ```suggestion
stack: ga 9.5
serverless: unavailable
```

applies_to:
  stack: ga
  serverless: ga
products:
  - id: kibana
  - id: cloud-serverless
---

# Select and switch Kibana spaces

The spaces you can access depend on your roles. Switching spaces does not change your permissions.

## Switch spaces

From the global header, select your current space, then select the space you want to open.

:::{image} /deploy-manage/images/kibana-change-space.png
:alt: Change current space menu
:screenshot:
:width: 50%
:::

## Understand what happens when you log in

{applies_to}`ech: ga` {applies_to}`serverless: ga` If you can access more than one space, {{kib}} asks you to select a space when you log in.

{applies_to}`self: ga` {applies_to}`ece: ga` {applies_to}`eck: ga` On self-managed deployments, what {{kib}} opens when you log in depends on your version and whether it has remembered a space:

   - {applies_to}`stack: ga 9.5+` {{kib}} can remember the space you use and return you to it. If {{kib}} has no remembered space or you no longer have access to it, {{kib}} asks you to select a space. A direct link continues to open the linked {{kib}} app or object instead of redirecting you to the remembered space.

>    - {applies_to}`stack: ga 9.0-9.4` If you can access more than one space, {{kib}} asks you to select a space.

Author: what I could test doesn't even strictly match this (ECH with 9.5 opened the default space regardless) - to confirm

## Change whether Kibana remembers your space
```{applies_to}
stack: ga 9.5
serverless: unavailable
```

This personal preference controls whether {{kib}} returns you to your last space or shows the space selector when you log in and can access multiple spaces.

**Requirements**:

> - Your Kibana instance must be self-managed or deployed through {{ece}} or {{eck}}. This menu option is not available on {{ecloud}}.

Author: ```suggestion
- Your Kibana instance must be self-managed or deployed through {{ece}} or {{eck}}. This option is not available on {{ecloud}}.
```
